### ISSUE TITLE
make info per leg more accessible

### ISSUE THREAD

**Issue by karussell** [graphhopper/graphhopper#1639]
Currently it is tricky to get e.g. distance per leg. (add distance until instruction with sign==5 or end is reached). The same happens with time etc. We could either use the "legs" array or introduce a new path detail like "way_points" so that we directly get the indices for the legs.

**Comment by richardthombs** [comment]
Yes please! I just started using the route api and quickly ran into this issue.

Additionally, the fact that you actually _can_ figure out per-leg information is not detailed in the documentation. I was lucky to find it on the news blog.
